commands/poll.js

```diff
@@ -0,0 +1,333 @@
+        dispose: true,
+      });
+
+      reactionCollector.on("collect", async (reaction, user) => {
+        console.info(`Got reaction ${reaction.emoji.name}  from ${user.tag}`);
+      });
+
+      reactionCollector.on("remove", async (reaction, user) => {
+        console.info(`Removed reaction ${reaction.emoji.name}  by ${user.tag}`);
+      });
+
+      reactionCollector.on("end", (collected) => {
+        console.debug(`Collected ${collected.size} reaction interactions!`);
+      });
+
+      const buttonCollector = pollMessage.createMessageComponentCollector({
+        componentType: ComponentType.Button,
+        time: pollCloseInSeconds * 1000,
+      });
+
+      buttonCollector.on("collect", async (action) => {
+        if (action.user.id === interaction.member.user.id) {
+          const originalEmbed = pollMessage.embeds[0];
+          const computingResultsEmbed = {
+            title: "Computing poll results...",
+            footer: {
+              text: "This bot is powered by Xander's money",
+            },
+          };
+
+          await pollMessage.edit({
+            embeds: [computingResultsEmbed],
+            components: [],
+          });
+
+          const finalResults = await getTopVotedOption(
+            pollMessage,
+            interaction
+          );
+
+          const finalEmbed = EmbedBuilder.from(originalEmbed)
+            .setDescription(
+              finalResults.length > 1
+                ? `Closed by ${interaction.member.user.tag}, see top options below!`
+                : `Closed by ${interaction.member.user.tag}, see top option below!`
+            )
+            .spliceFields(0, options.length)
+            .addFields(
+              finalResults.map((result) => {
+                const option = options.find(
+                  (item) => item.emoji === result.emoji
+                );
+
+                return {
+                  name:
+                    result.count > 1
+                      ? `${option.emoji}  - ${option.option} (${result.count} votes)`
+                      : `${option.emoji}  - ${option.option} (1 vote)`,
+                  value:
+                    result.count > 0
+                      ? `Voted by: ${result.reactors}`
+                      : `Voted by no one...`,
+                };
+              })
+            );
+
+          pollEnded = true;
+
+          await reactionCollector.stop(
+            `${interaction.member.user.tag} has opted to cancel the poll!`
+          );
+
+          await buttonCollector.stop(
+            `${interaction.member.user.tag} has opted to cancel the poll!`
+          );
+
+          await pollMessage.edit({
+            embeds: [finalEmbed],
+            components: [],
+          });
+        } else {
+          await action.reply({
+            content:
+              "Sorry, you can't cancel the poll since you did not create it!",
+            ephemeral: true,
+          });
+        }
```

Comment:
Query: Is it possible to use `setDisabled()` in `ButtonBuilder()` instead, where you could put something like `setDisabled(!interaction.member.user.id)`? This is so that people who did not create the poll can see the button on their end albeit disabled.

If it's not possible, then we can stick with the ephemeral part as I think this is okay.

Reply:
Sure, we can keep this.